dml.go

```diff
@@ -58,43 +74,63 @@ func doBatchedRequestsForCollection(auth authentication, method string, url stri
 
 		body, err := json.Marshal(payload)
 		if err != nil {
-			dmlErrors = errors.Join(dmlErrors, err)
-			break
+			return &results, err
 		}
 
 		resp, err := doRequest(method, url, jsonType, auth, string(body), http.StatusOK)
 		if err != nil {
-			dmlErrors = errors.Join(dmlErrors, err)
-			break
+			return &results, err
 		}
-		salesforceErrors := processSalesforceResponse(*resp)
-		if salesforceErrors != nil {
-			dmlErrors = errors.Join(dmlErrors, salesforceErrors)
+		_results, err := processSalesforceResponse(*resp)
+		if err != nil {
+			return &results, err
+		}
+
+		results.Results = append(results.Results, _results...)
```

Comment:
```suggestion
		currentResults, err := processSalesforceResponse(*resp)
		if err != nil {
			return &results, err
		}

		results.Results = append(results.Results, currentResults...)
```

We don't usually use underscore in Go variables